x/incentives/keeper/gauge_test.go

```diff
@@ -872,3 +985,60 @@ func (s *KeeperTestSuite) validateGaugeInfo(expected types.InternalGaugeInfo, ac
 		s.Require().Equal(expected.GaugeRecords[i].CumulativeWeight.String(), actual.GaugeRecords[i].CumulativeWeight.String())
 	}
 }
+
+// retrieves the gauge of expectedGauge.ID from state and validates that it matches expectedGauge
+func (s *KeeperTestSuite) validateGauge(expectedGauge types.Gauge) {
+	// Get gauge and check that the denom is set correctly
+	gauge, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, expectedGauge.Id)
+	s.Require().NoError(err)
+
+	// No denom set for group gauges.
+	s.Require().Equal(expectedGauge.DistributeTo.Denom, gauge.DistributeTo.Denom)
+	// ByGroup type set
+	s.Require().Equal(expectedGauge.DistributeTo.LockQueryType, gauge.DistributeTo.LockQueryType)
+	s.Require().Equal(expectedGauge.IsPerpetual, gauge.IsPerpetual)
+	s.Require().Equal(expectedGauge.Coins, gauge.Coins)
+	s.Require().Equal(expectedGauge.StartTime, gauge.StartTime.UTC())
+	s.Require().Equal(expectedGauge.NumEpochsPaidOver, gauge.NumEpochsPaidOver)
+}
+
+// test helper to create a gauge bypassing all checks and restrictions
+// It is useful in edge case tests that rely on invalid gagues written to store (e.g. in Distribute())
+func (s *KeeperTestSuite) createGaugeNoRestrictions(isPerpetual bool, coins sdk.Coins, distrTo lockuptypes.QueryCondition, startTime time.Time, numEpochsPaidOver uint64, poolID uint64) types.Gauge {
+	// Fund incentives module account to simulate transfer from owner to module account
+	s.FundModuleAcc(types.ModuleName, coins)
+	lastGaugeID := s.App.IncentivesKeeper.GetLastGaugeID(s.Ctx)
+	nextGaugeID := lastGaugeID + 1
+	gauge := types.Gauge{
+		Id:                nextGaugeID,
+		IsPerpetual:       isPerpetual,
+		Coins:             coins,
+		DistributeTo:      distrTo,
+		StartTime:         startTime,
+		NumEpochsPaidOver: numEpochsPaidOver,
+	}
+
+	if poolID != 0 {
+		s.App.PoolIncentivesKeeper.SetPoolGaugeIdNoLock(s.Ctx, poolID, nextGaugeID)
+	}
+
+	err := s.App.IncentivesKeeper.SetGauge(s.Ctx, &gauge)
+	s.Require().NoError(err)
+	s.App.IncentivesKeeper.CreateGaugeRefKeys(s.Ctx, &gauge, incentiveskeeper.CombineKeys(types.KeyPrefixUpcomingGauges, incentiveskeeper.GetTimeKeys(startTime)))
+	s.Require().NoError(err)
+
+	// Retrieve from state and return
+	gaugeFromState, err := s.App.IncentivesKeeper.GetGaugeByID(s.Ctx, nextGaugeID)
+	s.Require().NoError(err)
+	return *gaugeFromState
+}
+
+// validates that there is not gauge with the given ID in the slice
+func (s KeeperTestSuite) validateNoGaugeIDInSlice(slice []types.Gauge, gaugeID uint64) {
+	gaugeMatch := osmoutils.Filter(func(gauge types.Gauge) bool {
+		return gauge.Id == gaugeID
+	}, slice)
+	// No gauge matched ID.
+	s.Require().Empty(gaugeMatch)
+
+}
```

Comment:
Can ignore but this feels like we could maybe abstract this helper as a whole to osmoutils

Reply:
Probably not `osmoutils` because this depends on `apptesting.Suite`. I will move this to `apptesting` though, good idea!

Reply:
Will address in next PR